lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2895,6 +2895,31 @@ FIRRTLType impl::inferBitwiseResult(FIRRTLType lhs, FIRRTLType rhs,
   return UIntType::get(lhs.getContext(), resultWidth);
 }
 
+FIRRTLType impl::inferElementwiseResult(FIRRTLType lhs, FIRRTLType rhs,
+                                        std::optional<Location> loc) {
+  auto base = lhs.dyn_cast<FIRRTLBaseType>();
+  if (!base) {
+    if (loc)
+      mlir::emitError(*loc, "operands must have a base type");
+    return {};
+  }
+  // TODO: Remove following conditions if we start allowing element wise
```

Comment:
It seems like elementwise operations should have no more restrictions than their element counterpart